### ISSUE TITLE
react@16.5.0 but i use react@17.0.2

### ISSUE THREAD

**Issue by gxanshu** [artyorsh/react-native-eva-icons#45]
## 💬 Question

Could not resolve dependency:
npm ERR! peer react@"16.5.0" from react-native-eva-icons@1.3.1

**Comment by samuelcbird** [comment]
I have the same problem. Is there a fix for this?

**Comment by agjini** [comment]
I've also the same problem with my project that use react 18.
For the moment we have to use --legacy-peer-deps flag on npm install.

I'm proposing a PR where I'm trying to solve the issue : 

PR #46

**Comment by artyorsh** [comment]
The fix from @agjini resolves the issue, but for the sake of completeness I need to reinstall deps to regenerate `yarn.lock` - otherwise the codebase is not in actual state. I ignored it for a while, so now it will take time to bump dependencies and make sure everything, including example app, is runnable. I will post an update to NPM once this is done.

If you need a fix please either use `--legacy-peer-deps` or install the package from github

**Comment by agjini** [comment]
Hello @artyorsh and thanks for the explanation. Please let us know when the new version is published on npm.

**Comment by agjini** [comment]
Hi @artyorsh, any updates on npm publish ?

**Comment by artyorsh** [comment]
Yep. I'm currently working on it in my free time, please understand. The progress so far: https://github.com/artyorsh/react-native-eva-icons/pull/47

It seems the only thing to fix is jest setup, something to do with config files. The rest seem to be running normally

**Comment by Mioriarty** [comment]
For me, the error still occured when I tried to build my expo app using expo's eas cli. I just wanted to share how I fixed the issue. It is pretty simple:
1. Create a  file named `.npmrc` in the root directory
2. Write in that file the line `legacy-peer-deps=true`

I hope that helps some of you!

**Comment by SoloReverse** [comment]
On Expo projects, you could add the following in your package.json: 

`  "scripts": {
    "eas-build-pre-install": "npm config set legacy-peer-deps true",
  },`
